lib/dl_file_uploader_api_lib/dl_file_uploader_api_lib_tests/req_builder.py

```diff
@@ -167,29 +168,45 @@ def internal_params(
     @classmethod
     def update_conn_data(
         cls,
-        connection: GSheetsFileS3Connection,
+        connection: GSheetsFileS3Connection | YaDocsFileS3Connection,
         save: bool,
+        file_type: Optional[str] = "gsheets",
         *,
         require_ok: bool = True,
     ) -> Req:
         sources_desc = [src.get_desc() for src in connection.data.sources]
+        sources = list()
+        if file_type == "gsheets":
+            sources = [
+                dict(
+                    id=src_desc.source_id,
+                    title=src_desc.title,
+                    spreadsheet_id=src_desc.spreadsheet_id,
+                    sheet_id=src_desc.sheet_id,
+                    first_line_is_header=src_desc.first_line_is_header,
+                )
+                for src_desc in sources_desc
+            ]
+        elif file_type == "yadocs":
+            sources = [
+                dict(
+                    id=src_desc.source_id,
+                    title=src_desc.title,
+                    public_link=src_desc.public_link,
+                    sheet_id=src_desc.sheet_id,
+                    first_line_is_header=src_desc.first_line_is_header,
+                )
+                for src_desc in sources_desc
+            ]
         return Req(
             method="post",
             url="/api/v2/update_connection_data",
             data_json={
                 "connection_id": connection.uuid,
                 "authorized": False,
                 "save": save,
-                "sources": [
-                    dict(
-                        id=src_desc.source_id,
-                        title=src_desc.title,
-                        spreadsheet_id=src_desc.spreadsheet_id,
-                        sheet_id=src_desc.sheet_id,
-                        first_line_is_header=src_desc.first_line_is_header,
-                    )
-                    for src_desc in sources_desc
-                ],
```

Comment:
Will this work if we just feed a list of `get_desc`'s results here? These file-type branches are starting to get annoying